### PR TITLE
Add type to onClick event listener

### PR DESCRIPTION
**Pull request by gtupak**
We could do something like this to avoid using `any` types. I didn't spend too much time trying to fix the test so I skipped it, but we could fix it in the next pr

### CODE REVIEW

File: test/ArcxAnalyticsSdk.test.ts [hunk 1]
```diff
@@ -467,7 +467,7 @@ describe('(unit) ArcxAnalyticsSdk', () => {
         expect(eventStub).to.not.have.been.called
       })
 
-      it('catches click event', () => {
+      xit('catches click event', () => {
```
Comment (gtupak) [inline test/ArcxAnalyticsSdk.test.ts:470]: To be unskipped in the next pr? or if it's too difficult, the test "makes track clicks on element without defined classname and id" in `ArcxAnalyticsProvider.test.tsx` can be enough

Comment (xitronix) [reply test/ArcxAnalyticsSdk.test.ts]: I rebased and made some changes :) So that's your turn to review :)

**Review by gtupak — COMMENTED** [review]
lgtm, discussed irl

**Review by xitronix — APPROVED** [review]

**Pull request merged**